Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+        } catch (Exception $e) {
+          throw $e;
+        }
+      }
+      // Any other case (e.g. value is integer) is ok as it is.
+
+      $where .= $clause[0].' '.$oprt.' '.$value;
+    }
+
+    $this->where = $where;
+  }
+
+  /**
+   * Correct value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @static
+   *
+   * @param mixed $value Scalar value.
+   * @param string $method Name of the method that called this one (directly or
+   * indirectly).
+   * @return mixed The correct value.
+   *
+   * @throws \RuntimeException if the value is an array.
+   */
+  private static function correctValue($value, $method) {
+    if (is_string($value)) {
+      // Put quotes around the value if it is a string.
+      $value = "'".$value."'";
+    } elseif (is_null($value)) {
+      // Replace the value with a string 'NULL' if it is null.
+      $value = 'NULL';
+    } elseif (is_bool($value)) {
+      // Replace the value with a string 'TRUE' or 'FALSE' if it is boolean.
+      $value = ($value == true) ? 'TRUE' : 'FALSE';
+    } elseif (!is_scalar($value)) {
+      throw new RuntimeException(
+        'Arguments passed to `Query::'.$method.'()` aren\'t in the correct form'
+      );
+    }
+    // Any other case (e.g. value is integer) is ok as it is.
+    return $value;
+  }
+
+  /**
+   * Verify the operator.
+   *
+   * Checks that the operator passed is a valid one.
+   *
+   * @since 1.0.0
+   * @access private
+   * @static
+   *
+   * @param string $operator Operator that needs to be verified.
+   * @param string $method Name of the method that called this one.
+   * @return boolean Whether the operator is valid or not.
+   */
+  private static function isOperatorValid($operator, $method) {
+    if ($method == 'join')
+      $valid_operators = array('=', '>', '>=', '<', '<=', '!=', 'like', 'LIKE');
+    else
+      $valid_operators = array('=', '>', '>=', '<', '<=', '!=', 'between', 'BETWEEN', 'like', 'LIKE');
+
+    return in_array($operator, $valid_operators);
+  }
+
+  /**
+   * Build SELECT query string.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @return string
+   *
+   * @throws \RuntimeException if $table property isn't set.
+   */
+  private function buildSelect() {
+    // Create the `SELECT` part.
+    $query = 'SELECT ';
+    if (!is_array($this->select) || empty($this->select)) {
+      $part = $this->buildCount();
+      if ($part == '')
+        $query .= '* ';
+      else
+        $query .= $part.' ';
+    }
+    else {
+      $count = 0;
+      foreach ($this->select as $field) {
+        if ($count != 0)
+          $query .= ', ';
+        $query .= $field;
+        $count++;
+      }
+      $part .= $this->buildCount();
+      if ($part != '')
+        $query .= ', '.$part;
+      $query .= ' ';
+    }
+
+    // Append the `FROM` part.
+    $query .= 'FROM ';
+    if (!isset($this->table))
+      throw new RuntimeException('Cannot execute the query without a target table');
```

Comment:
You can move this check at the beginning of the method, so that you avoid doing other stuff and than throwing the exception.
Also, you should check if `$this->table` is `null`, not `isset()`.